### ISSUE TITLE
ta-lib 2 X DEMA custom strategy failed

### ISSUE THREAD

**Issue by orangekj** [askmike/gekko#1298]
**Note: for support questions, please join our [Discord server](https://discord.gg/26wMygt)**

* **I'm submitting a ...**
[ ] bug report
[X] feature request
[ ] question about the decisions made in the repository

* **Action taken** (what you did)
Hi guys,

I would like to create strategy to calculate diff on short & long DEMA.
I thought it would be easy, just add few codes to built-in DEMA but by looking at the source it was different than the DEMA formula I know.

DEMA = ( 2 * EMA(n)) - (EMA(EMA(n)) ), where n= period

So I leverage to use ta-lib DEMA but result shows that it is not working as expected.

* **Expected result** (what you hoped would happen)

calculate diff on short & long DEMA

* **Actual result** (unexpected outcome)

no calculation and no trade on backtest.

* **Other information** (e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. stackoverflow, etc)

Since I'm newbie on JAVA, it is impossible to I hope someone can help on this :-(

`// custom settings:
// config.custom = {
//  short: "optInTimePeriod: 9",
//  long: "optInTimePeriod: 18"
// }

config['talib-dema'] = {
  parameterShort: {
    optInTimePeriod: 9,
  },
  parameterLong: {
    optInTimePeriod: 18,
  },
  thresholds: {
    down: -0.025,
    up: 0.025,
  }
}
`
To get Short & Long DEMA result, I've used parameterShort and parameterLong.
Not sure this value will be readable from strategy.

`// If you want to use your own trading methods you can
// write them here. For more information on everything you
// can use please refer to this document:
//
// https://github.com/askmike/gekko/blob/stable/docs/trading_methods.md

// helpers
var _ = require('lodash');
var log = require('../core/log.js');

// Let's create our own method
var method = {};

// Prepare everything our method needs
method.init = function() {
  this.name = 'talib-dema'
  // keep state about the current trend
  // here, on every new candle we use this
  // state object to check if we need to
  // report it.
  this.trend = 'none';

  // how many candles do we need as a base
  // before we can start giving advice?
  this.requiredHistory = this.tradingAdvisor.historySize;

//  var customShortDEMASettings = this.settings.short;
//  var customLongDEMASettings = this.settings.long;
  var customShortDEMASettings = this.settings.parameterShort;
  var customLongDEMASettings = this.settings.parameterLong;

  // define the indicators we need
  this.addTalibIndicator('dema1', 'dema', customShortDEMASettings);
  this.addTalibIndicator('dema2', 'dema', customLongDEMASettings);

}

// What happens on every new candle?
method.update = function(candle) {
  // nothing!
}

// for debugging purposes: log the last calculated
// DEMAs and diff.
//  var dema = this.indicators.dema;

method.log = function() {

  log.debug('calculated DEMA properties for candle:');
  log.debug('\t', 'long DEMA:', this.shortresult); //.toFixed(8));
  log.debug('\t', 'short DEMA:', this.longresult); //.toFixed(8));
  log.debug('\t diff:', this.demadiff); //.toFixed(5));
//  log.debug('\t DEMA age:', this.shortresult.age, 'candles');

}

// Based on the newly calculated
// information, check if we should
// update or not.
method.check = function(candle) {
  var price = candle.close;
  var shortresult = this.talibIndicators.dema1.result.outreal;
  var longresult = this.talibIndicators.dema2.result.outreal;
  var demadiff = 100 * (shortresult - longresult) / ((shortresult + longresult) / 2);
// var macddiff = result['outMACD']

  var message = '@ ' + price.toFixed(8) + ' (' + demadiff.toFixed(5) + ')';

  if(this.settings.thresholds.down > demadiff && this.trend !== 'short') {
    this.trend = 'short';
    this.advice('short');

  } else if(this.settings.thresholds.up < demadiff && this.trend !== 'long'){
    this.trend = 'long';
    this.advice('long');

  }
}

module.exports = method;`

Basically, I copied talib-macd.js to create talib-dema.js.
Made some modification to calc short DEMA and long DEMA.
Added debug log code copied from DEMA.js (tofixed is not working so //)

There is no error but backtest result is

![image](https://user-images.githubusercontent.com/33338494/32535443-6649dd4a-c49d-11e7-9faf-c914502b6472.png)

**Comment by orangekj** [comment]
[talib-dema.txt](https://github.com/askmike/gekko/files/1452864/talib-dema.txt)
[config.txt](https://github.com/askmike/gekko/files/1452865/config.txt)

**Comment by Gab0** [comment]
@orangekj You can modify native DEMA strategy to the formula you like, no need to mess with evil talib.

**Comment by Niekk68** [comment]
@orangekj You have to set the "R" as capital.
var shortresult = this.talibIndicators.dema1.result.out**R**eal;
var longresult = this.talibIndicators.dema2.result.out**R**eal;

**Comment by askmike** [comment]
Assuming this is fixed, if not please comment here :)
